src/preferences/PreferencesBase.js

```diff
@@ -173,6 +174,14 @@ define(function (require, exports, module) {
                 prefFile.read({}, function (err, text) {
                     if (err) {
                         if (createIfNew) {
+                            // Unreadable file is also unwritable -- delete so get recreated
+                            if (err === FileSystemError.NOT_READABLE || err === FileSystemError.UNSUPPORTED_ENCODING) {
+                                appshell.fs.moveToTrash(path, function (err) {
```

Comment:
It doesn't seem right introducing `FileSystemError` and `appshell.fs` here. Maybe a new `File.prototype.moveToTrash()` method?

Are these the right errors to look for? WHat about `FileSystemError.NOT_SUPPORTED` ?